server/src/main/java/org/opensearch/index/translog/TranslogManager.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+/**
+ * The interface that orchestrates Translog operations and manages the {@link Translog} and interfaces with the Engine
+ *
+ * @opensearch.internal
+ */
+public interface TranslogManager {
+
+    /**
+     * Rolls the translog generation and cleans unneeded.
+     */
+    void rollTranslogGeneration() throws TranslogException;
```

Comment:
The implementation of methods in `TranslogManager` will revolve around Translog reference it is holding, right? Or each implementation of `TranslogManager` have a different implementation of these methods.